### PR TITLE
Improving toPlatformCase

### PR DESCRIPTION
**Pull request by say25**
I saw that the helpful `.toPlatformCase` was added as part of https://github.com/desktop/desktop/pull/5744/. Shout out to @Daniel-McCarthy. 

This Pull Request:
1. Adds MOAR Unit Tests
2. Add the ability to ignore "default" platform case rules to handle cases like those in [Context-Menu.ts](https://github.com/desktop/desktop/blob/master/app/src/ui/lib/context-menu.ts)

### CODE REVIEW

File: app/test/unit/platform-case-test.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { expect } from 'chai'
 import { toPlatformCase } from '../../src/lib/platform-case'
 
 describe('string to platform case', () => {
-  it('convert string to detected platform text case', () => {
+  it('converts lower string to detected platform text case', () => {
```
Comment (iAmWillShepherd) [inline app/test/unit/platform-case-test.ts]: s/lower/lowercase

**Comment by say25** [comment]
@iAmWillShepherd makes sense to me. I was trying to improve the test coverage an noticed the same thing you did which is the whole situation with words like `in` and `and`. Given the request for future discussion we should put this on hold it would seem.

**Comment by say25** [comment]
Depending on the outcome of the discussion, we can backout the extra commits.

**Comment by iAmWillShepherd** [comment]
> @iAmWillShepherd makes sense to me. I was trying to improve the test coverage an noticed the same thing you did which is the whole situation with words like in and and. 

Yea I figured and I initially took it as that until I really honed in on the function you're testing. I think you were on the right track with considering test coverage, but I think we may be removing the function you exercised because as you just said, it can get complicated in a hurry! 👍

**Comment by shiftkey** [comment]
> Given the request for future discussion we should put this on hold it would seem.

I'm going to close this out while we discuss this further in #5841 - thanks for digging into this in the meantime @say25!

**Pull request closed without merging**